### PR TITLE
backupccl: enable probabilistic MT testing in helpers

### PR DESCRIPTION
**Pull request by adityamaru**
Informs: https://github.com/cockroachdb/cockroach/issues/115640
Release note: None

### CODE REVIEW

File: pkg/ccl/backupccl/full_cluster_backup_restore_test.go [hunk 1]
```diff
@@ -54,63 +55,77 @@ func TestFullClusterBackup(t *testing.T) {
...
+						JobDisablePersistingCheckpoints: true,
+					},
+					GCJob: &sql.GCJobTestingKnobs{
+						// We want to run the GC job to completion without waiting for
+						// MVCC GC.
+						SkipWaitingForMVCCGC: true,
+					},
+				}}}
+		const numAccounts = 10
+		// TODO(adityamaru): We need to ensure that both these clusters have the same
```
Comment (msbutler) [inline pkg/ccl/backupccl/full_cluster_backup_restore_test.go:94]: this todo can be removed, yeah?

File: pkg/ccl/backupccl/full_cluster_backup_restore_test.go [hunk 2]
```diff
@@ -54,63 +55,77 @@ func TestFullClusterBackup(t *testing.T) {
...
-	_, sqlDB, tempDir, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, numAccounts, InitManualReplication, params)
-	tcRestore, sqlDBRestore, cleanupEmptyCluster := backupRestoreTestSetupEmpty(t, singleNode, tempDir, InitManualReplication, params)
-	defer cleanupFn()
-	defer cleanupEmptyCluster()
+	// It is easier to assert state in this test if both the backing up and
+	// restoring cluster are run with the same tenant option. We already have
+	// targeted tests for combinations of backing up and restoring between a
+	// system and application tenant.
+	var runOnceUnderDuress bool
+	testutils.RunTrueAndFalse(t, "inSystemTenant", func(t *testing.T, inSystemTenant bool) {
```
Comment (msbutler) [inline pkg/ccl/backupccl/full_cluster_backup_restore_test.go:63]: do we need to run this test under both configs for every test run? I believe each case would take many seconds to run. I don't feel strongly about this though.

**Review by msbutler — APPROVED** [review]
Thank you thank you thank you for doing this!!!

File: pkg/ccl/backupccl/full_cluster_backup_restore_test.go (hunk 1 above)
Comment (adityamaru) [reply pkg/ccl/backupccl/full_cluster_backup_restore_test.go:94]: indeed

File: pkg/ccl/backupccl/full_cluster_backup_restore_test.go (hunk 2 above)
Comment (adityamaru) [reply pkg/ccl/backupccl/full_cluster_backup_restore_test.go:63]: I added logic to run it once under duress cause as you said this takes multiple minutes. Under non-duress runs thought its not too bad, and probably worth the extra coverage.

**Comment by adityamaru** [comment]
TFTR!

bors r=msbutler

**Comment by adityamaru** [comment]
blathers backport 23.2

**Pull request merged**